src/cpu/matmul/ref_matmul.cpp

```diff
@@ -221,41 +235,118 @@ status_t ref_matmul_t::execute_ref(const exec_ctx_t &ctx) const {
+    dim_t N_chunk = std::max<dim_t>(2, dst_scale_group_n);
+    parallel_nd(batch, utils::div_up(M, M_chunk), utils::div_up(N, N_chunk),
+            [&](dim_t mb, dim_t mc, dim_t nc) {
+                for_(dim_t m_ = mc * M_chunk;
+                        m_ < std::min<dim_t>((mc + 1) * M_chunk, M);
+                        m_ += dst_scale_group_m)
+                for (dim_t n_ = nc * N_chunk;
+                        n_ < std::min<dim_t>((nc + 1) * N_chunk, N);
+                        n_ += dst_scale_group_n) {
+                    float max_dst_group = 0.0f;
+                    for_(dim_t m_gidx = 0; m_gidx < dst_scale_group_m; m_gidx++)
+                    for (dim_t n_gidx = 0; n_gidx < dst_scale_group_n;
+                            n_gidx++) {
+                        dim_t m = m_ + m_gidx;
+                        dim_t n = n_ + n_gidx;
+                        dims_t dst_dims_idx;
+                        const size_t offset = mb * M * N + m * N + n;
+                        utils::l_dims_by_l_offset(
+                                dst_dims_idx, offset, dst_d.dims(), ndims);
+
+                        float d = ker(dst_dims_idx, m, n);
+                        if (bias) d += ker_bias(dst_dims_idx);
+
+                        const auto dst_off = dst_d.off_v(dst_dims_idx);
+                        if (non_default_attrs) {
+                            if (with_dropout)
+                                d = ref_dropout(
+                                        d, dropout_mask, dst_off, *p, *seed);
+                            ref_post_ops_t::args_t args;
+                            args.dst_val = io::load_float_value(
+                                    sum_dt, dst, dst_off);
+                            args.ctx = &ctx;
+                            args.l_offset = offset;
+                            args.dst_md = pd()->dst_md();
+                            ref_post_ops->execute(d, args);
+                        }
+                        if (attr_scales.get(DNNL_ARG_DST).is_dynamic()) {
+                            max_dst_group = std::max(max_dst_group, ::fabsf(d));
+                            io::store_float_value(
+                                    data_type::f32, d, temp_dst, dst_off);
+                        } else {
+                            if (with_dst_scales) d *= dst_scales[0];
+                            if (dst_rnd_mode == rounding_mode::stochastic)
+                                d = math::stochastic_round_fwd(d, dst_off,
+                                        rnd_seed[0], dst_d.data_type());
+                            io::store_float_value(
+                                    dst_d.data_type(), d, dst, dst_off);
+                            utils::dim_iterator(
+                                    dst_d.dims(), dst_dims_idx, batch_ndims);
+                        }
+                    }
+
+                    if (attr_scales.get(DNNL_ARG_DST).is_dynamic()) {
```

Comment:
Is it supposed to e a maximum within the processed group or within the tensor?

Reply:
Maximum for the processed group. This is why it is computed inside the loop nest.